### ISSUE TITLE
contentful_rails requires contentful_model, but doesn't.

### ISSUE THREAD

**Issue by TrevorBramble** [errorstudio/contentful_rails#5]
Unless the `contentful_model` gem is added to the Gemfile or a require statement is inserted at the top of the `contentful_rails.rb` initializer, Rails bombs.

If it's a non-optional dependency, please require `contentful_model` from the `contentful_rails` gem.

```
$ rails s
=> Booting WEBrick
=> Rails 4.2.3 application starting in development on http://localhost:3000
=> Run `rails server -h` for more startup options
=> Ctrl-C to shutdown server
Exiting
/home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/contentful_rails-0.2.2.1/lib/contentful_rails/engine.rb:13:in `block in <class:Engine>': uninitialized constant ContentfulRails::En
gine::ContentfulModel (NameError)                                                                                                                                                                      from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/initializable.rb:30:in `instance_exec'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/initializable.rb:30:in `run'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/initializable.rb:55:in `block in run_initializers'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:226:in `block in tsort_each'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:429:in `each_strongly_connected_component_from'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:347:in `block in each_strongly_connected_component'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:345:in `each'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:345:in `call'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:345:in `each_strongly_connected_component'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:224:in `tsort_each'
        from /home/trevor/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/tsort.rb:203:in `tsort_each'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/initializable.rb:54:in `run_initializers'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/application.rb:352:in `initialize!'
        from /home/trevor/code/chef/www-contentful-rails/config/environment.rb:5:in `<top (required)>'
        from /home/trevor/code/chef/www-contentful-rails/config.ru:3:in `require'
        from /home/trevor/code/chef/www-contentful-rails/config.ru:3:in `block in <main>'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/builder.rb:55:in `instance_eval'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/builder.rb:55:in `initialize'
        from /home/trevor/code/chef/www-contentful-rails/config.ru:in `new'
        from /home/trevor/code/chef/www-contentful-rails/config.ru:in `<main>'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/builder.rb:49:in `eval'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/builder.rb:49:in `new_from_string'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/builder.rb:40:in `parse_file'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/server.rb:299:in `build_app_and_options_from_config'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/server.rb:208:in `app'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands/server.rb:61:in `app'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/rack-1.6.4/lib/rack/server.rb:336:in `wrapped_app'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands/server.rb:139:in `log_to_stdout'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands/server.rb:78:in `start'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:80:in `block in server'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:75:in `tap'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:75:in `server'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/railties-4.2.3/lib/rails/commands.rb:17:in `<top (required)>'
        from /home/trevor/code/chef/www-contentful-rails/bin/rails:8:in `require'
        from /home/trevor/code/chef/www-contentful-rails/bin/rails:8:in `<top (required)>'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/spring-1.3.6/lib/spring/client/rails.rb:28:in `load'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/spring-1.3.6/lib/spring/client/rails.rb:28:in `call'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/spring-1.3.6/lib/spring/client/command.rb:7:in `call'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/spring-1.3.6/lib/spring/client.rb:26:in `run'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/spring-1.3.6/bin/spring:48:in `<top (required)>'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/spring-1.3.6/lib/spring/binstub.rb:11:in `load'
        from /home/trevor/.rvm/gems/ruby-2.2.2@www-contentful-rails/gems/spring-1.3.6/lib/spring/binstub.rb:11:in `<top (required)>'
        from /home/trevor/code/chef/www-contentful-rails/bin/spring:13:in `require'
        from /home/trevor/code/chef/www-contentful-rails/bin/spring:13:in `<top (required)>'
        from bin/rails:3:in `load'
        from bin/rails:3:in `<main>'
```

**Comment by TrevorBramble** [comment]
Ah, and now that I've added the initializer for `contentful_model`, it turns out that it isn't enough to add a require like so.
```ruby
# config/initializers/contentful_rails.rb
require 'contentful_model'
```
Either it has to be added to the above _and_ to `config/initializers/contentful_model.rb` or the `contentful_model` gem has to be added directly to the project Gemfile.

**Comment by edtjones** [comment]
Hi @TrevorBramble - if you're using a Rails application and you're included the ContentfulRails gem in the project, it should install ContentfulModel for you - it's in the gemspec here: https://github.com/errorstudio/contentful_rails/blob/master/contentful_rails.gemspec

Also - no need to `require contentful_model` in the initializer - Rails will autoload ContentfulModel.

Ed

**Comment by edtjones** [comment]
(Closing - reopen if I'm mistaken :-) )

**Comment by TrevorBramble** [comment]
@edtjones Hi, Ed. Right, declaring the dependency in the Gemspec file will make sure it's installed, but Rails doesn't know to autoload it because it relies on Bundler for that. Without contentful_model in the Gemfile, Rails doesn't load it automatically.

Frankly, I admire the instinct to keep these libraries distinct, but since they're mutually dependent it seems in practice that's not actually useful here. =^)

**Comment by edtjones** [comment]
A really good point @TrevorBramble - sorry. I messed up there.

I still it's worth keeping the distinction between the model stuff and the rails stuff - because it's quite possible (and we've done it) that you want to use Contentful models in something non-railsy. And the Rails stuff adds loads of hard deps on the Rails stack - caching, engines, initialisers etc.

Shall I add an initialiser in the contentful_rails engine to `require()` it? Seems like the logical solution. (Or please feel free to do that and send a PR).

**Comment by TrevorBramble** [comment]
That's fantastic--looking again there's only the ActiveSupport requirement, not all of Rails, nor the contentful_rails gem. So that's good because I'm also evaluating Contentful for use with Sinatra and Interactions without any sort of content presentation.

How's #8 for the `require`?

**Comment by edtjones** [comment]
Sorry Trevor - missed this. Need to work on my github notification settings. Will review the PR ASAP and merge.

**Comment by zrisher** [comment]
This can be closed now that you've merged #8, correct?

**Comment by edtjones** [comment]
Yes - closing. Thanks for the nudge @zrisher
